### PR TITLE
Prepping for 2.6.0 release

### PR DESCRIPTION
**Pull request by sejli**
### Description
- Bumping plugin version to `2.6.0`
   - **Note:** The `main` branch should point to `3.0.0`. Following the release, we will fix our branches to point to the correct versions.
- Incrementing `jackson-core` to `2.14.2`
- Adding release notes
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [x] Commits are signed as per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/search-processor/blob/main/CONTRIBUTING.md).

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -53,7 +53,7 @@ validateNebulaPom.enabled = false
 
 buildscript {
     ext {
-        opensearch_version = "2.5.0"
+        opensearch_version = "2.6.0"
```
Comment (noCharger) [inline build.gradle:56]: should it be 3.0 on main?

File: release-notes/search-processor.release-notes-2.6.0.md [hunk 2]
```diff
@@ -0,0 +1,13 @@
+## Version 2.6.0 Release Notes
+
+### Enhancements
+* Add more test coverage [#105](https://github.com/opensearch-project/search-processor/pull/105)
```
Comment (noCharger) [inline release-notes/search-processor.release-notes-2.6.0.md]: Use PR from main branch https://github.com/opensearch-project/search-processor/pull/100

Same for the rest

File: helpers/search_processing_kendra_quickstart.sh [hunk 3]
```diff
@@ -7,7 +7,7 @@ set -o nounset
 
 # Some useful constants
 readonly DOCKER_IMAGE_TAG="opensearch-with-ranking-plugin"
-readonly OPENSEARCH_VERSION="2.5.0"
```
Comment (msfroh) [inline helpers/search_processing_kendra_quickstart.sh:10]: Note that we will need to update the quickstart script's template for the plugin zip to reflect the new naming convention. (Of course, until there's a 2.6.0 plugin zip attached to the release, it doesn't matter.)

Comment (msfroh) [reply helpers/search_processing_kendra_quickstart.sh:10]: Specifically on line 23, we'll need to make it something like `opensearch-search-processor-2.6.0.zip`, I think.

**Review by msfroh — APPROVED** [review]

**Review by noCharger — APPROVED** [review]
Approve with comment - Let's create another issue to track version issue post release

**Review by msfroh — APPROVED** [review]

**Pull request merged**